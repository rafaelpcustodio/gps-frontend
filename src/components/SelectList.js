import PropTypes from 'prop-types'
import React from 'react'

import SelectItem from './SelectItem'
import StyledSelectList from './StyledSelectList'
import If from '../components/If'
import { isNotEmpty } from '../utils/functions'

const SelectList = ({ expanded, onSelect, options }) => {
    
    return(
        <StyledSelectList expanded={ expanded }>
            { options && options.map((option, index) =>
                <SelectItem 
                    key={ index } 
                    onClick={ onSelect } 
                    option={ option }
                />)
            }
        </StyledSelectList>
    )
}

SelectList.defaultProps = {
    expanded: false,
    onSelect: null,
    options: []
}

SelectList.propTypes = {
    expanded: PropTypes.bool.isRequired,
    onSelect: PropTypes.func.isRequired,
    options: PropTypes.arrayOf(PropTypes.string).isRequired
}

export default SelectList
