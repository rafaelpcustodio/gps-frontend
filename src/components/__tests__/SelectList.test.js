import configureMockStore from 'redux-mock-store'
import { Map, Record } from 'immutable'
import React from 'react'
import renderer from 'react-test-renderer'

import { Provider } from 'react-redux'

import SelectList from '../SelectList'

describe('Tests for SelectList component', () => {
    let store = {}
    let mockStore = {}

    it('Should render without crash', () => {
        mockStore = configureMockStore()
        store = mockStore(Record({
            gpsReducers: Map({
            })
        })())

        const tree = renderer.create(
            <Provider store={ store }>
                <SelectList
                    expanded={true}
                    options={['']}
                    onSelect={ jest.fn() }
                />
            </Provider>
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })
})
