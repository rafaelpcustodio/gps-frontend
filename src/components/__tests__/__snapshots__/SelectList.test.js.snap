// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Gps component Should render without crash 1`] = `
Array [
  <div
    className="SearchBarContainer-sc-1byzrjc-0 janIPg"
  >
    <input
      className="LicensePlate__StyledLicensePlateText-pf7qoz-0 ksfRmt"
      maxLength="7"
      onChange={[Function]}
      placeholder="License plate (ex: ABC1234)"
      value=""
    />
    <div
      className="StyledSelect-sc-17xe3wx-0 cnxjXs"
      disabled={true}
    >
      <div
        className="StyledSelectedContainer__StyledSelectContainer-sc-2m0ndn-0 fwyTFq"
      >
        <span
          className="SelectPoiTypeContainer__SelectText-sc-19t5xdi-0 iSHPxC"
          disabled={true}
          onClick={[Function]}
        >
          Select POI type
        </span>
        <button
          className="StyledDropdown-sc-1d17mlt-0 crWdhk"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-caret-down fa-w-10 fa-lg "
            color="#aaaaaa"
            data-cy="icon"
            data-icon="caret-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </div>
      <div
        className="StyledSelectList-vv0ppr-0 djStfM"
      >
        <li
          className="SelectItem__StyledSelectItem-c7nnhy-0 lkKUay"
          onClick={[Function]}
        >
          Restaurants
        </li>
        <li
          className="SelectItem__StyledSelectItem-c7nnhy-0 lkKUay"
          onClick={[Function]}
        >
          Hotels
        </li>
      </div>
    </div>
    <div
      className="StyledSelect-sc-17xe3wx-0 cnxjXs"
      disabled={true}
    >
      <div
        className="StyledSelectedContainer__StyledSelectContainer-sc-2m0ndn-0 fwyTFq"
      >
        <span
          className="SelectRadiusContainer__SelectText-sz6f5m-0 heeAuE"
          disabled={true}
          onClick={[Function]}
        >
          Select radius
        </span>
        <button
          className="StyledDropdown-sc-1d17mlt-0 crWdhk"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-caret-down fa-w-10 fa-lg "
            color="#aaaaaa"
            data-cy="icon"
            data-icon="caret-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </div>
      <div
        className="StyledSelectList-vv0ppr-0 djStfM"
      >
        <li
          className="SelectItem__StyledSelectItem-c7nnhy-0 lkKUay"
          onClick={[Function]}
        >
          1000
        </li>
        <li
          className="SelectItem__StyledSelectItem-c7nnhy-0 lkKUay"
          onClick={[Function]}
        >
          2000
        </li>
        <li
          className="SelectItem__StyledSelectItem-c7nnhy-0 lkKUay"
          onClick={[Function]}
        >
          4000
        </li>
        <li
          className="SelectItem__StyledSelectItem-c7nnhy-0 lkKUay"
          onClick={[Function]}
        >
          4500
        </li>
      </div>
    </div>
    <button
      className="StyledButton-sc-1z0vb0n-0 ejBCHp"
      onClick={[Function]}
    >
      <div
        className="Button__StyledText-ko51rl-0 eowQDx"
      >
        Apply
      </div>
    </button>
  </div>,
  <div
    className="StyledMinimumDistance__StyledMinimumDisance-vqza48-0 kYkUOE"
  >
    Closest opened Service: 400m
  </div>,
  <div>
    Loading...
  </div>,
]
`;

exports[`Tests for Gps component Should render without crash 2`] = `
Array [
  <div
    className="SearchBarContainer-sc-1byzrjc-0 janIPg"
  >
    <input
      className="LicensePlate__StyledLicensePlateText-pf7qoz-0 ksfRmt"
      maxLength="7"
      onChange={[Function]}
      placeholder="License plate (ex: ABC1234)"
      value=""
    />
    <div
      className="StyledSelect-sc-17xe3wx-0 cnxjXs"
      disabled={true}
    >
      <div
        className="StyledSelectedContainer__StyledSelectContainer-sc-2m0ndn-0 fwyTFq"
      >
        <span
          className="SelectPoiTypeContainer__SelectText-sc-19t5xdi-0 iSHPxC"
          disabled={true}
          onClick={[Function]}
        >
          Select POI type
        </span>
        <button
          className="StyledDropdown-sc-1d17mlt-0 crWdhk"
        >
          <
            className=""
            color="#aaaaaa"
            data-cy="icon"
            icon={
              Object {
                "icon": Array [
                  320,
                  512,
                  Array [],
                  "f0d7",
                  "M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z",
                ],
                "iconName": "caret-down",
                "prefix": "fas",
              }
            }
            size="lg"
          />
        </button>
      </div>
      <div
        className="StyledSelectList-vv0ppr-0 djStfM"
      >
        <li
          className="SelectItem__StyledSelectItem-c7nnhy-0 lkKUay"
          onClick={[Function]}
        >
          Restaurants
        </li>
        <li
          className="SelectItem__StyledSelectItem-c7nnhy-0 lkKUay"
          onClick={[Function]}
        >
          Hotels
        </li>
      </div>
    </div>
    <div
      className="StyledSelect-sc-17xe3wx-0 cnxjXs"
      disabled={true}
    >
      <div
        className="StyledSelectedContainer__StyledSelectContainer-sc-2m0ndn-0 fwyTFq"
      >
        <span
          className="SelectRadiusContainer__SelectText-sz6f5m-0 heeAuE"
          disabled={true}
          onClick={[Function]}
        >
          Select radius
        </span>
        <button
          className="StyledDropdown-sc-1d17mlt-0 crWdhk"
        >
          <
            className=""
            color="#aaaaaa"
            data-cy="icon"
            icon={
              Object {
                "icon": Array [
                  320,
                  512,
                  Array [],
                  "f0d7",
                  "M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z",
                ],
                "iconName": "caret-down",
                "prefix": "fas",
              }
            }
            size="lg"
          />
        </button>
      </div>
      <div
        className="StyledSelectList-vv0ppr-0 djStfM"
      >
        <li
          className="SelectItem__StyledSelectItem-c7nnhy-0 lkKUay"
          onClick={[Function]}
        >
          1000
        </li>
        <li
          className="SelectItem__StyledSelectItem-c7nnhy-0 lkKUay"
          onClick={[Function]}
        >
          2000
        </li>
        <li
          className="SelectItem__StyledSelectItem-c7nnhy-0 lkKUay"
          onClick={[Function]}
        >
          4000
        </li>
        <li
          className="SelectItem__StyledSelectItem-c7nnhy-0 lkKUay"
          onClick={[Function]}
        >
          4500
        </li>
      </div>
    </div>
    <button
      className="StyledButton-sc-1z0vb0n-0 ejBCHp"
      onClick={[Function]}
    >
      <div
        className="Button__StyledText-ko51rl-0 eowQDx"
      >
        Apply
      </div>
    </button>
  </div>,
  <div
    className="StyledMinimumDistance__StyledMinimumDisance-vqza48-0 kYkUOE"
  >
    Closest opened Service: 400m
  </div>,
  <div>
    Loading...
  </div>,
]
`;

exports[`Tests for SelectList component Should render without crash 1`] = `
<div
  className="StyledSelectList-vv0ppr-0 ifTzoY"
>
  <li
    className="SelectItem__StyledSelectItem-c7nnhy-0 lkKUay"
    onClick={[Function]}
  >
    error
  </li>
</div>
`;
